nnvm/src/top/tensor/reduce.cc

```diff
@@ -322,6 +322,64 @@ values over a given axis.
       topi::argmin(inputs[0], axis, param.keepdims) };
 });
 
+NNVM_REGISTER_REDUCE_OP(mean)
+  .describe(R"code(Computes the mean of array elements over given axes.
+
+Example::
+
+  data = [[[1,2],[2,3],[1,3]],
+          [[1,4],[4,3],[5,2]],
+          [[7,1],[7,2],[7,3]]]
+
+  mean(data, axis=1)
+  [3.22]
+
+  mean(data, axis=[1,2])
+  [ 2.  3.16666667  4.5]
+
+)code" NNVM_ADD_FILELINE)
+.set_attr<FTVMCompute>(
+  "FTVMCompute", [](const NodeAttrs& attrs,
+                    const Array<Tensor>& inputs,
+                    const Array<Tensor>& out_info) {
+    const ReduceParam& param = nnvm::get<ReduceParam>(attrs.parsed);
+    TShape r_axes = GetReduceAxes(inputs[0]->shape.size(),
+                                  param.axis, param.exclude);
+    if (!r_axes.ndim()) return Array<Tensor> { topi::identity(inputs[0]) };
```

Comment:
Maybe we can follow MXNet implementation for mean:
The default, axis=(), will compute over all elements into a scalar array with shape (1,).
If axis is int, a reduction is performed on a particular axis.
If axis is a tuple of ints, a reduction is performed on all the axes specified in the tuple.
If exclude is true, reduction will be performed on the axes that are NOT in axis instead.
Negative values means indexing from right to left.

Reply:
Now the behavior is the same as MXNet.

Reply:
@srkreddy1238 what do you mean support weights?

Reply:
There would be additional weight param which is multiplied to input before mean calculation.
Ref. https://www.tensorflow.org/api_docs/python/tf/metrics/mean

This was a last dependency for resnet50 
Ref. https://discuss.tvm.ai/t/anyone-successful-converting-tensorflow-resnet50-model-to-nnvm/766/4

I will work on enhancing the current mean to support weighted version.

Reply:
Hi @srkreddy1238, 
I think nnvm top should be similar to numpy api? 
Maybe it's better to add weight support to tensorflow convertor, because weighted version is TF specific after all? 
What do you think?

Reply:
@yuruofeifei  agreed :+1: 

I think weight multiple can be handled at front end by broadcast multiply followed by normal mean.